bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/HomeAssistantChannelTransformation.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mqtt.homeassistant.internal;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.mqtt.homeassistant.internal.component.AbstractComponent;
+import org.openhab.core.thing.binding.generic.ChannelTransformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hubspot.jinjava.Jinjava;
+import com.hubspot.jinjava.interpret.FatalTemplateErrorsException;
+
+/**
+ * Provides a channel transformation for a Home Assistant channel with a
+ * Jinja2 template, providing the additional context and extensions required by Home Assistant
+ * Based in part on the JinjaTransformationService
+ *
+ * @author Cody Cutrer - Initial contribution
+ */
+@NonNullByDefault
+public class HomeAssistantChannelTransformation extends ChannelTransformation {
+    private final Logger logger = LoggerFactory.getLogger(HomeAssistantChannelTransformation.class);
+
+    private final Jinjava jinjava;
+    private final AbstractComponent component;
+    private final String template;
+
+    public HomeAssistantChannelTransformation(Jinjava jinjava, AbstractComponent component, String template) {
+        super((String) null);
+        this.jinjava = jinjava;
+        this.component = component;
+        this.template = template;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return template.isEmpty();
+    }
+
+    @Override
+    public Optional<String> apply(String value) {
+        String transformationResult;
+        Map<String, @Nullable Object> bindings = new HashMap<>();
+
+        logger.debug("about to transform '{}' by the function '{}'", value, template);
+
+        bindings.put("value", value);
+
+        try {
+            JsonNode tree = new ObjectMapper().readTree(value);
```

Comment:
I think you can re-use the `ObjectMapper` instead of creating a new instance on every call.